internal/jobcontainers/logon.go

```diff
@@ -0,0 +1,64 @@
+package jobcontainers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/winapi"
+	"golang.org/x/sys/windows"
+)
+
+// Takes in a DOMAIN\Username or just Username and will return a token
+// for the account if successful.
+func processToken(user string) (windows.Token, error) {
+	var (
+		domain   string
+		userName string
+		token    windows.Token
+	)
+
+	split := strings.Split(user, "\\")
+	if len(split) == 2 {
+		domain = split[0]
+		userName = split[1]
+	} else if len(split) == 1 {
+		userName = split[0]
+	}
+
+	// If empty or ContainerUser or ContainerAdministrator just let it inherit the token
+	// from whatever is used to launch it (containerd-shim etc). Regular container images
+	// are usable for jobcontainers so we need to handle the ContainerUser and ContainerAdministrator
+	// cases also.
+	if user == "" || user == "ContainerUser" || user == "ContainerAdministrator" {
+		return openCurrentProcessToken()
+	}
+
+	var logonType uint32
+	// User asking to run as a local system account (NETWORK SERVICE, LOCAL SERVICE, SYSTEM)
+	if domain == "NT AUTHORITY" {
+		logonType = winapi.LOGON32_LOGON_SERVICE
+	} else {
+		logonType = winapi.LOGON32_LOGON_INTERACTIVE
+	}
+
+	// They want a user account. Password accounts not supported
```

Comment:
Nit: I feel like this should go at the beginning of the function but I'm not tied to it